### PR TITLE
[TIMOB-25672] iOS: Fix core-SDK warnings, leaks and possible crashes

### PR DESCRIPTION
**Pull request by hansemannn**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25672

### CODE REVIEW

**Review by vijaysingh-axway — CHANGES_REQUESTED** [review]
@hansemannn Can you look in the issue? Thanks!

File: iphone/Classes/TiMediaVideoPlayerProxy.m [hunk 1]
```diff
@@ -186,7 +188,6 @@ - (void)viewWillAttach
 
 - (void)viewDidDetach
 {
-  [self removeNotificationObserver];
```
[the remark this answers is not in the document]
Comment (hansemannn) [reply iphone/Classes/TiMediaVideoPlayerProxy.m]: Jenkins actually passes the tests, but fails later. Which test did it mention?

[the remark this answers is not in the document]
Comment (hansemannn) [reply iphone/Classes/TiMediaVideoPlayerProxy.m]: Just reproduced it, adding a test-case for it in my test-suite as well.

[the remark this answers is not in the document]
Comment (hansemannn) [reply iphone/Classes/TiMediaVideoPlayerProxy.m]: Fixed as part of https://github.com/appcelerator/titanium_mobile/pull/9733/commits/31182c65d25b6a7d134790aa5571912c1ad5983d

**Review by vijaysingh-axway — APPROVED** [review]
CR passed.

**Comment by mukherjee2** [comment]
Passed FR using this environment:
Node Version: 8.9.1
NPM Version: 5.5.1
Mac OS: 10.13.1
Appc CLI: 7.0.1
Appc CLI NPM: 4.2.11
Titanium SDK version: 7.0.2 (artifacts from this PR)
Appcelerator Studio vers 5.0.0
iOS 11.1

Validated by building and launching app on device. I observed no warnings or memory leaks using Instruments.

**Review by mukherjee2 — APPROVED** [review]
Passed FR. See comments.

**Pull request merged**